myia/public_api.py

```diff
@@ -603,7 +613,8 @@ def reshape(x, *shp):
 def scatter(self, dim, index, src):
     """Map of 'scatter' pytorch method."""
     if not P.hastype(src, APT):
```

Comment:
Consider using this instead of APT:

```python
AA = AbstractArray(
    AbstractScalar({TYPE: ANYTHING, VALUE: ANYTHING}),
    {SHAPE: ANYTHING, TYPE: ANYTHING},
)
```

The `TYPE: ANYTHING` should make it work for any kind of array, whether it's an ndarray or a Tensor.

Reply:
this suggestion doesn't seem to necessarily work.